orderer/common/cluster/deliver_test.go

```diff
@@ -1451,3 +1451,56 @@ func TestBlockPuller_UpdateEndpoint(t *testing.T) {
 		require.True(t, couldNotConnectLogged)
 	})
 }
+
+func newBlockFetcher(dialer *countingDialer, orderers ...string) *cluster.BlockFetcher {
+	cbp := cluster.BlockPuller{
+		Dialer:              dialer,
+		Channel:             "mychannel",
+		Signer:              &mocks.SignerSerializer{},
+		Endpoints:           endpointCriteriaFromEndpoints(orderers...),
+		FetchTimeout:        time.Second * 10,
+		MaxTotalBufferBytes: 1024 * 1024, // 1MB
+		RetryTimeout:        time.Millisecond * 10,
+		VerifyBlockSequence: noopBlockVerifierf,
+		Logger:              flogging.MustGetLogger("test"),
+		StopChannel:         make(chan struct{}),
+	}
+	return &cluster.BlockFetcher{
+		BlockPuller:       cbp,
+		BFTTimeOutChannel: make(chan struct{}),
+	}
+}
+
+// TODO: Test writing for BlockFetcher is in progress.
+// names of the functions and structure may change during development.
+func TestBlockFetcherBasicHappyPath(t *testing.T) {
+	// Scenario: Single ordering node,
+	// and the block puller pulls blocks 5 to 10.
+	osn := newClusterNode(t)
+	defer osn.stop()
+
+	dialer := newCountingDialer()
+	bp := newBlockFetcher(dialer, osn.srv.Address())
+
+	// The first seek request asks for the latest block
+	osn.addExpectProbeAssert()
```

Comment:
but if you have a block puller dependency as an interface why do you need to mock an entire deliver service? You can just mock the interface no?

Reply:
Entire deliver service is mocked to be consistent with the unit-tests  lock puller. I will explore the possibility to mock the interface only.